doc/model.rst

```diff
@@ -0,0 +1,683 @@
+parameters to a file or attachment a file of parameters with values to the
+Model entity for use within the simulation.
+Users can specify whether to copy, symlink, or configure files by passing parameters
+one of the following parameters: `to_copy`, `to_symlink`, or `to_configure`.
+It's worth noting that invoking
+this function, ``attach_generator_files()``, post-attachment overwrites the existing list of entity files.
+
+To write Model parameters to a file and attach to the Model, pass one of the
+following values to the helper function:
+
+* `to_copy` (list, optional): Files that are copied into the path of the entity.
+* `to_symlink` (list, optional): Files that are symlinked into the path of the entity.
+
+To attach a file to the Model and read the contents for use within the simulation, pass the
+following value to the helper function:
+
+* `to_configure` (list, optional): Designed for text-based model input files,
+  "to_configure" is exclusive to models. These files contain parameters for
+  the model, with customizable tags corresponding to values users intend to
+  modify. The default tag is a semicolon (e.g., THERMO = ;10;).
+
+In the `Example` subsection, we provide an example illustrating using the value `to_configure`
+within ``attach_generator_files()``.
+
+-------
+Example
+-------
+This example provides a demonstration of how to instruct SmartSim to attach a file
+containing parameters for use within the Model using ``Model.attach_generator_files()``
+and specifying `to_configure`.
+
+We have a text file named `configure_inputs.txt`. Within the text, is the parameter, `THERMO`,
+that is assigned a single value:
+
+.. code-block:: txt
+
+   THERMO = ;10;
+
+The parameter `THERMO` is used within our application script and we would like to
+instruct SmartSim to assign the parameter value upon Model execution.
+
+To encapsulate our simulation using a Model, we must create an Experiment instance
+to gain access to the Experiment helper function that creates the Model.
+Begin by importing the Experiment module and SmartSim log module.
+Initialize an Experiment:
+
+.. code-block:: python
+
+    from smartsim import Experiment
+    from smartsim.log import get_logger
+
+    logger = get_logger("Experiment Log")
+    # Initialize the Experiment
+    exp = Experiment("getting-started", launcher="auto")
+
+Models require run settings. Create a simple ``RunSettings`` object, specifying
+the path to our application script and the executable to run the script:
+
+.. code-block:: python
+
+    # Initialize a RunSettings object
+    model_settings = exp.create_run_settings(exe="python", exe_args="/path/to/application.py")
+
+Next, initialize a ``Model`` object with ``Experiment.create_model()``
+and pass in the `model_settings` instance:
+
+.. code-block:: python
+
+    # Initialize a Model object
+    model = exp.create_model("model", model_settings)
+
+We now have a ``Model`` instance named `model`. Attach the above text file
```

Comment:
nit: having the name and variable match names could be confusing in followups "instance named `model`" could refer to the name or variable. Consider swapping the model name to something unique for minor differentiation, e.g. "my-model"?